code-challenges.test.js

```diff
@@ -36,16 +46,78 @@ const object3 = { number: -7 }
+// output: number checked if evenly divisible by 3 or not
+// create a function using object as the paramentor
+// create a conditional statement that evaluates whether the object.number is evenly divisble by 3 or not
+// returns true or false
+
+// const div3 = (number) => {
+//     if (number % 3 === 0) {
+//         return true
+//     } else {
+//         return false
+//     }
+// }
+
+// I redid this one in the shorter form for practice.
+
+const div3 = (number) => number % 3 === 0
+
+console.log(div3(object1.number))
+
+
+
+
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("capWords", () => {
+    it('takes in an array of words and returns an array with all the words capitalized', () => {
+        expect(capWords(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(capWords(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+})
+
+// Good Fail: ReferenceError: capWords is not defined
+
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
\ No newline at end of file
+// Pseudo code:
+// Create a function using array as parameter
+// Have .map iterate through the array of strings
+// Use .split("") method to make the array of strings into array of letters
+// Have .map() iterate through the array letters
+// Use a conditional statement to evaluate the first index [0]
+// Using .toUpperCase() to capitalize it
+// Use .join("") method to restore the arrays of letters back into an array of strings
+
+const capWords = (array) => {
+    return array.map((string) => {
+       return string.split("").map((letter, index) => {
+            if (index === 0){
+                return letter.toUpperCase()
+            } else {
+                return letter
+            }
+       }).join("")
+    })
+}
+console.log(capWords(randomNouns1))
+
+// Had one of my friends who is a senior javascript developer look at this because I wanted a greater understanding of .filter() and I wanted to make sure it couldn't be applied. She explained .filter() can only be truth or falsey statements, and I couldn't run two different conditions like I do with the if/else conditional statement. She did however mention trying .slice() instead. This is what I came up with.
+
+// const capWords = (array) => {
+//     return array.map((string) => {
+//         return string.slice(0,1).toUpperCase() + string.slice(1)
+//     })
+// }
+
+// I think .slice() makes this look a lot cleaner.
```

Comment:
Yep just remember the basics! `.filter` will return a subset of the original array, because it will only return the values that meet the filter criteria. `.map` will return an array of the same length as the original, with logic applied to each value. Remembering this, you would be able to deduce that you don't want to use filter for this challenge.